### ISSUE TITLE
Logging not working with custom environment

### ISSUE THREAD

**Issue by stompyfox** [werner-duvaud/muzero-general#73]
I made a custom gym environment using the cartpole.py file as a template with minimal changes yet logging breaks for some reason. It just prints:

`Last test reward: 0.00. Training step: 0/10000. Played games: 0. Loss: 0.00`

every time. Any help on this would be highly appreciated.

Edit: The agent in question still seems to learn properly.

**Comment by ahainaut** [comment]
Hi,
Does the games of the repository works fine for you ?  If so, the error is probably from your environment.
Could you be more precise about the reasons why you think the agent seems to learn, please ?

**Comment by stompyfox** [comment]
> Does the games of the repository works fine for you ?

If you mean the preinstalled game environments like cartpole then yes.

> Could you be more precise about the reasons why you think the agent seems to learn, please ?

The agent seems to improve on the environment. If I set learning rate to zero it does substantially worse.

**Comment by ahainaut** [comment]
Ok, so the issue is probably in the implementation of your environment, you can test it with the option `Test the game manually`. I suggest you visit our [Discord page](https://discord.com/invite/GB2vwsF) to try to find out more about the issues you may be having with a custom environment. We would like to keep Github issues for bugs or features related to the code, and Discord for questions about custom environments.

**Comment by stompyfox** [comment]
I posted my question on the discord. I didn't do anything unusual when making my environment. I simply followed this tutorial: https://medium.com/@apoddar573/making-your-own-custom-environment-in-gym-c3b65ff8cdaa

**Comment by ahainaut** [comment]
Ok, I suggest we continue the conversation on the Discord page !

**Comment by werner-duvaud** [comment]
For the record,
Logs and training start only after that at least one game has been selfplayed.
So if the number of moves per episode is really high or if the environment is not returning done=True after a few steps, the logs will hang.

For issues related to your own custom environment, please join and ask on our dedicated [discord channel](https://discord.gg/GB2vwsF).
